crates/orchestrator/src/jobs/types.rs

```diff
@@ -138,6 +140,57 @@ pub struct JobItem {
     pub updated_at: DateTime<Utc>,
 }
 
+/// Defining a structure that contains the changes to be made in the job object,
+/// id and created at are not allowed to be changed
+// version and updated_at will always be updated when this object updates the job
+#[derive(Serialize, Debug)]
+pub struct JobItemUpdates {
+    pub internal_id: Option<String>,
+    pub job_type: Option<JobType>,
+    pub status: Option<JobStatus>,
+    pub external_id: Option<ExternalId>,
+    pub metadata: Option<HashMap<String, String>>,
+}
+
+/// implements default selections for JobItemUpdates
+impl Default for JobItemUpdates {
+    fn default() -> Self {
+        JobItemUpdates { internal_id: None, job_type: None, status: None, external_id: None, metadata: None }
+    }
+}
+
+/// implements only needed singular changes
+impl JobItemUpdates {
+    pub fn only_status(status: JobStatus) -> Self {
+        JobItemUpdates { status: Some(status), ..Default::default() }
+    }
+    pub fn only_metadata(metadata: HashMap<String, String>) -> Self {
+        JobItemUpdates { metadata: Some(metadata), ..Default::default() }
+    }
+    pub fn to_document(&self, current_job: &JobItem) -> Result<Document> {
+        let mut doc = Document::new();
+
+        // Serialize the struct to BSON
+        let bson = bson::to_bson(&self)?;
+
+        // If serialization was successful and it's a document
+        if let Bson::Document(bson_doc) = bson {
```

Comment:
if this doesn't work should we throw an error?

Reply:
`
let bson = bson::to_bson(&self)?;`
using the ? we are already throwing the error into a `color_eyre` report 

`if let Bson::Document(bson_doc) = bson {`
Bson is an enum, off which Document is a embedded type,

What else is needed ?
What am I missing here ? @apoorvsadana 


<img width="345" alt="Screenshot 2024-10-07 at 3 18 39 PM" src="https://github.com/user-attachments/assets/de5e5d91-9028-4122-8f78-1b2b52ce8779">
<img width="854" alt="Screenshot 2024-10-07 at 3 19 01 PM" src="https://github.com/user-attachments/assets/47657680-87c1-4aba-84b1-f3c130a85380">

Reply:
i mean what if it's not a document? we still update version and updated_at?

Reply:
Valid, resolved [here](https://github.com/madara-alliance/madara-orchestrator/pull/134/commits/881b3aac10f025ffe5e81fa15f80703e57c667dc).